pgmpy/inference/Sampling.py

```diff
@@ -0,0 +1,196 @@
+        >>> inference.forward_sample(2)
+                diff       intel       grade
+        0  (diff, 1)  (intel, 0)  (grade, 1)
+        1  (diff, 1)  (intel, 0)  (grade, 2)
+        """
+        sampled = DataFrame(index=range(size), columns=self.topological_order)
+        for node in self.topological_order:
+            cpd = self.cpds[node]
+            if cpd.evidence:
+                weights = []
+                for i in range(size):
+                    evid = [sampled[var][i] for var in cpd.evidence]
+                    weights.append(cpd.reduce(evid, inplace=False).values)
+                sampled[node] = sample_discrete(cpd.variables[cpd.variable], weights)
+            else:
+                # can generate the column at once
+                weights = cpd.values
+                sampled[node] = sample_discrete(cpd.variables[cpd.variable], weights, size)
+        return sampled
+
+    def rejection_sample(self, evidence=None, size=1):
+        """
+        Generates sample(s) from joint distribution of the bayesian network,
+        given the evidence.
+
+        Parameters
+        ----------
+        evidence: dict
+            a dict key, value pair as {var: state_of_var_observed}
+            None if no evidence
+        size: int
+            size of sample to be generated
+
+        Returns
+        -------
+        sampled: pandas.DataFrame
+            the generated samples
+
+        Examples
+        --------
+        >>> from pgmpy.models.BayesianModel import BayesianModel
+        >>> from pgmpy.factors.CPD import TabularCPD
+        >>> from pgmpy.factors.Factor import State
+        >>> from pgmpy.inference.Sampling import BayesianModelSampling
+        >>> student = BayesianModel([('diff', 'grade'), ('intel', 'grade')])
+        >>> cpd_d = TabularCPD('diff', 2, [[0.6], [0.4]])
+        >>> cpd_i = TabularCPD('intel', 2, [[0.7], [0.3]])
+        >>> cpd_g = TabularCPD('grade', 3, [[0.3, 0.05, 0.9, 0.5], [0.4, 0.25,
+        ...                0.08, 0.3], [0.3, 0.7, 0.02, 0.2]],
+        ...                ['intel', 'diff'], [2, 2])
+        >>> student.add_cpds(cpd_d, cpd_i, cpd_g)
+        >>> inference = BayesianModelSampling(student)
+        >>> evidence = {'diff': State(var='diff', state=0)}
+        >>> inference.rejection_sample(evidence, 2)
+                intel       diff       grade
+        0  (intel, 0)  (diff, 0)  (grade, 1)
+        1  (intel, 0)  (diff, 0)  (grade, 1)
+        """
+        if evidence is None:
+            return self.forward_sample(size)
+        sampled = DataFrame(columns=self.topological_order)
+        prob = 1
+        while len(sampled) < size:
+            _size = int((size - len(sampled)) / prob)
+            _sampled = self.forward_sample(_size)
+            for i in range(_size):
+                for var in evidence:
+                    if evidence[var] != _sampled[var][i]:
+                        _sampled.drop(i, inplace=True)
+                        break
```

Comment:
Using ```break``` statement is not a very clean solution. So instead of using a ```for``` loop with an conditional ```if``` and ```break``` could you please change it to ```while``` loop. It would be much cleaner and easier to read as well.

@ankurankan Your suggestions?